src/test/scala/org/bblfsh/client/v2/BblfshClientParseTest.scala

```diff
@@ -0,0 +1,71 @@
+package org.bblfsh.client.v2
+
+import java.nio.ByteBuffer
+
+import gopkg.in.bblfsh.sdk.v2.protocol.driver.ParseResponse
+import org.scalatest.{BeforeAndAfter, BeforeAndAfterAll, FlatSpec, Matchers}
+
+import scala.io.Source
+
+class BblfshClientParseTest extends FlatSpec
+  with BeforeAndAfter
+  with BeforeAndAfterAll
+  with Matchers {
+
+  val client = BblfshClient("0.0.0.0", 9432)
+  val fileName = "src/test/resources/SampleJavaFile.java"
+  val fileContent = Source.fromFile(fileName).getLines.mkString("\n")
+  var resp: ParseResponse = _
+
+  import BblfshClient._ // enables uast.* methods
+
+  before {
+    resp = client.parse(fileName, fileContent)
+  }
+
+  override def afterAll {
+    client.close()
+  }
+
+  "Parsed UAST for .java file" should "not be empty" in {
+    assert(resp != null)
+    assert(resp.errors.isEmpty)
+  }
+
+  "Decoded UAST after parsing" should "not be NULL" in {
+    val uast = resp.uast.decode()
+
+    assert(uast != null)
+    assert(uast.nativeContext != 0)
+
+    println(uast)
+    uast.dispose()
+  }
+
+  "Decoded UAST RootNode" should "not be NULL" in {
+    val uast = resp.uast.decode()
+    val rootNode: Node = uast.root()
+    println(rootNode.getClass)
+
+    rootNode should not be null
```

Comment:
Can we also assert the structure, at least for the root node? That it has specific keys with specific types, for example. Checking the whole tree is even better :wink:

Reply:
Sure, we will improve tests as we move on.

BTW, could you help me understand what do you mean by `assert the structure` here, without th `load()` or any iterators?

Reply:
Yep, sure. The Nodes API works symmetrically for both client-hosted and `libuast`-hosted nodes, so it's possible to call methods like `Kind` to assert a type of a UAST node or get the list of keys the same way. You are right, it doesn't make sense to go deep before the `load` is implemented, but asserting few basic things might help to catch some bugs that may be triggered before the `load`.

Reply:
> asserting few basic things ...  call methods like Kind to assert a type of a UAST node or get the list of keys

@dennwc would you also be so kind to show me equivalent of this in https://github.com/bblfsh/client-python/ ? Thanks!

Reply:
No. Because it seems there isn't :sweat_smile: 

Opened a https://github.com/bblfsh/client-python/issues/159 to track this. Thanks for checking it!

Reply:
> The Nodes API works symmetrically for both client-hosted and libuast-hosted nodes, so it's possible to call methods like Kind to assert a type of a UAST node 

@dennwc Could you please point me to an example, how this might have looked in Python API?
On my reading of CPython extension - [PyNodeExt type](https://github.com/bblfsh/client-python/blob/master/bblfsh/pyuast.cc#L48) does not expose "kind", or am I looking at the wrong place?

Reply:
There is definitely a C API for this, but I remember now that the kind was not exposed in C++ API due to some issues with C++ templates.

So it seems like a bigger issue, let's address it in the following PRs.